api/controllers/game/set-is-ranked.js

```diff
@@ -0,0 +1,23 @@
+module.exports = async function (req, res) {
+  try {
+    const { isRanked } = req.body;
+
+    const game = await gameService.findGame({ gameId: req.session.game });
+  
+    const gameUpdates = { isRanked };
+
+    Game.publish([game.id], {
+      change:'setIsRanked',
+      userId: req.session.usr,
+      pNum: req.session.pNum,
+      gameId: game.id,
+      isRanked: isRanked,
+    });
+    await Game.updateOne({ id: game.id }).set(gameUpdates);
```

Comment:
Let's update the database before the socket update so that database errors are caught in the response before we send the socket update
```suggestion
    await Game.updateOne({ id: game.id }).set(gameUpdates);
    Game.publish([game.id], {
      change:'setIsRanked',
      userId: req.session.usr,
      pNum: req.session.pNum,
      gameId: game.id,
      isRanked: isRanked,
    });
```